### ISSUE TITLE
Cannot read .split() of undefined in the google analytics plugin

### ISSUE THREAD

**Issue by kixxauth** [oddnetworks/oddworks#54]
Getting this stack trace:

```
Unhandled rejection TypeError: Cannot read property 'split' of undefined
    at Object.prepare (services/oddworks-master/services/events/analyzers/google-analytics.js:19:32)
    at Object.send (services/oddworks-master/services/events/analyzers/google-analytics.js:43:18)
    at _.each.analyzer (services/oddworks-master/services/events/index.js:21:14)
    at arrayEach (services/oddworks-master/node_modules/lodash/index.js:1289:13)
    at Function.<anonymous> (services/oddworks-master/node_modules/lodash/index.js:3345:13)
    at config.bus.observe.payload (services/oddworks-master/services/events/index.js:19:5)
    atservices/oddworks-master/node_modules/oddcast/lib/channel.js:174:14
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

**Comment by blainsmith** [comment]
Ah that must be the Roku sending events with a malformed payload. We can gracefully handle this and just log that warning instead.

**Comment by barthdamon** [comment]
Are we sure it is only roku? Just double checked, the format of the request is what it always has been and the responses are coming back as 201. I'm going to update the ios sample app to use oddworks 2.0 with the new sdk. In doing so I will check out whether oddworks is still throwing this error for ios as well.

**Comment by blainsmith** [comment]
This came from John playing with Oddworks + Roku.

**Comment by barthdamon** [comment]
Right but @kixxauth mentioned he noticed the same error at one point, I'm guessing not using a roku? @pmcconnell have you run into this testing the iOS sdk yet? It could be just the roku, but not sure yet

**Comment by pmcconnell** [comment]
I saw some split errors earlier when things weren't configured correctly. Not in that module though. I don't see those errors now. (at least lately)

**Comment by kixxauth** [comment]
Guys, it's spit() not split() haha

Anyway, I think it was Erik who actually saw this before and left a comment about it in slack, so maybe this is all resolved. Let's leave it open for another day or two and see what happens.

**Comment by lorismaz** [comment]
I got the same error from my tvos app. For me the payload sent from the swift command `OddMetricService.postAppInitMetric()` looks like this: 

```json
{"attributes":{"action":"app:init"},"type":"event"}
```
so it should be 

`const action = payload.attributes.action.split(':');` 

instead of 

`const action = payload.action.split(':');` no ?

**Comment by kixxauth** [comment]
The google analytics plugin will be deprecated in v3

**Comment by kixxauth** [comment]
Fixed in 8880a607fbc8dca859aa2188931a1055b3f4dfca
